### DISCUSSION TITLE
Retrieving Objectbricks

### DISCUSSION THREAD

**Discussion by ManoDharmalingam** [pimcore/pimcore#15049] [Q&A / Support]
Hi ,
I tried creating the Object brick  to fetch the Objectbricks created and tried to retrieve the objectbricks using the steps provided in the below document
**https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Object_Bricks.html#page_Retrieving-Objectbricks**
As soon as I return the response, I receive the following error message
**Call to undefined method getBricks in class Pimcore\Model\DataObject\Products**

Please let me know how to resolve this error.

Regards,
Mano Dharmalingam.

**Comment by graceman9** [comment]
@ManoDharmalingam Have you found a solution?

**Reply by AlephLau** [reply]
@graceman9 The method getBricks() doesn't work, you must call the brick by its respective name. For example, if you added the brick to a product class, you should call $product = Product::getById(1); $brick = $product->getNameOfBrick(); and if you do a dd($product) you will be able to see the fields of your brick.

**Comment by g-weiler** [comment]
Hello, 
i'm also new to pimcore, but i found out that the name of the getter function for the objectbrick depends on the attributename that you have assigned to the objectbrick in the main class. 

For the example in the documentation you see the attributename is "attributes" , so in the PHP API you will have to use getAttributes() method instead of getBricks()

Hope this helps,
